dotenv.go

```diff
@@ -0,0 +1,218 @@
+package dotenv
+
+import (
+	"bufio"
+	"bytes"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"regexp"
+	"strings"
+	"unicode"
+)
+
+const (
+	posKey int = iota
+	posVal
+)
+
+// ErrInvalidln indicates an invalid line
+var ErrInvalidln = errors.New("invalid line")
+
+// ErrEmptyln indicates an empty line
+var ErrEmptyln = errors.New("empty line")
+
+// ErrCommentln indicates a comment line
+var ErrCommentln = errors.New("comment line")
+
+var varRE = regexp.MustCompile(`\${\w+}`)
+
+// ReadFile reads an env file at a given path, and return values as a map.
+func ReadFile(path string) (map[string]string, error) {
+	f, err := os.Open(path)
+	if err != nil {
+		return nil, err
+	}
+	defer f.Close()
+	return Read(f)
+}
+
+// Read parses the given reader's contents and return values as a map.
+func Read(rd io.Reader) (map[string]string, error) {
+	r := bufio.NewReader(rd)
+	envMap := make(map[string]string)
+	var (
+		line, k, v string
+		err        error
+	)
+	for {
```

Comment:
A `bufio.Scanner` might be easier and possibly more efficient here, but this is fine.

Reply:
I went through some research to try and find the most appropriate one. Which package is Scanner from? It doesn't seem to be bufio

Reply:
Never mind, I see it now in a different file in the package.